roles/nuage_master/files/serviceaccount.sh

```diff
@@ -59,5 +66,31 @@ oadm create-api-client-config --certificate-authority=${CA_CERT} --client-dir=${
 # Verify the finalized kubeconfig
 if ! [ $(oc whoami --config=$CONFIG_FILE) == 'system:serviceaccounts:default:nuage' ]; then
     echo "Service account creation failed!"
-    exit 1
 fi
+
+
+openssl genrsa -out $REST_CA_KEY 4096
+openssl req -new -x509 -key $REST_CA_KEY -out $REST_CA_CERT -subj "/CN=nuage-signer"
+echo '00' > $REST_CA_SERIAL
+
+OPENSSL_CONFIG='
+[ clientauth ]
+basicConstraints=CA:FALSE
+extendedKeyUsage=critical,clientAuth
+'
+
+echo "$OPENSSL_CONFIG" > ${OUTDIR}/openssl.cnf
+openssl genrsa -out $REST_SERVER_KEY 4096
+openssl req -key $REST_SERVER_KEY -new -out ${OUTDIR}/restServer.req -subj "/CN=$(hostname -f)"
+openssl x509 -req -in ${OUTDIR}/restServer.req -CA $REST_CA_CERT -CAkey \
+    $REST_CA_KEY -CAserial $REST_CA_SERIAL \
+    -out $REST_SERVER_CERT
+
+openssl genrsa -out $REST_CLIENT_KEY 4096
+openssl req -key $REST_CLIENT_KEY -new -out ${OUTDIR}/restClient.req -subj "/CN=nuage-client"
+openssl x509 -req -in ${OUTDIR}/restClient.req -CA $REST_CA_CERT -CAkey \
+    $REST_CA_KEY -CAserial $REST_CA_SERIAL \
+    -out $REST_CLIENT_CERT -extensions clientauth \
+    -extfile ${OUTDIR}/openssl.cnf
```

Comment:
This will create a CA on each of the nuage_master hosts, instead I believe you only want to create the CA on a single master and distribute the created certs to the other master and node hosts.

We are currently refactoring the way we are generating and distributing certificates. Currently the logic to generate and distribute them is done within the playbooks, but we are pushing that logic down into the roles. You can see the approach we are moving to here: https://github.com/openshift/openshift-ansible/pull/150

In general, you will most likely want the following:

- a nuage_ca role that creates the certificate authority to be used and uses delegate_to and run_once to ensure that the ca is created only on the delegated host (most likely you'll want groups.oo_first_master.0 to be the delegate_to host).
- a nuage_certificate role that creates the server and/or client certs based on which is needed and distributes them to the appropriate host. This role should use delegate_to to create the certificates on the CA host and local_action to facilitate transferring the certificates to the intended host. This role should also have a dependency defined on the nuage_ca role.
- modify the nuage_master and nuage_node roles to have a dependency on the nuage_certificate role, and to set the variable needed to tell it to create and distribute the server/client cert appropriately.